### PR TITLE
Allow to build tutorials out of project

### PR DESCRIPTION
**Pull request by clementperon**

### CODE REVIEW

File: Examples/Tutorials/Tutorial-HelloWorld/CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,17 @@
+cmake_minimum_required(VERSION 3.12)
+project(Tutorial-HelloWorld)
+
 add_executable(Tutorial-HelloWorld main.cpp)
 
-target_link_libraries(Tutorial-HelloWorld PUBLIC PcapPlusPlus::Pcap++)
+if(NOT TARGET PcapPlusPlus::Pcap++)
+        message("Compiling Tutorial out of project")
+        find_package(PcapPlusPlus REQUIRED)
```
Comment (seladb) [inline Examples/Tutorials/Tutorial-HelloWorld/CMakeLists.txt:8]: I ran it locally on Windows and Linux. It works great on Linux but on Windows if fails to find `Threads::Threads`.

When I added `find_package(Threads REQUIRED)` it worked perfectly on both platforms

File: Examples/Tutorials/Tutorial-HelloWorld/CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,17 @@
+cmake_minimum_required(VERSION 3.12)
```
Comment (seladb) [inline Examples/Tutorials/Tutorial-HelloWorld/CMakeLists.txt:1]: I think we can duplicate this for the rest of the tutorials

Comment (clementperon) [reply Examples/Tutorials/Tutorial-HelloWorld/CMakeLists.txt:1]: done

File: Examples/Tutorials/Tutorial-HelloWorld/CMakeLists.txt (hunk 1 above)
Comment (clementperon) [reply Examples/Tutorials/Tutorial-HelloWorld/CMakeLists.txt:8]: thanks for the catch, could you pull the latest commit, rebuild and reinstall PcapPP and then retry to build the tutorial out-of-project.

Comment (seladb) [reply Examples/Tutorials/Tutorial-HelloWorld/CMakeLists.txt:8]: It works now, thanks!

File: Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+add_executable("${PROJECT_NAME}" main.cpp)
 
-set_target_properties(Tutorial-DpdkL2Fwd PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${PCAPPP_BINARY_TUTORIAL_DIR}")
+if(NOT TARGET PcapPlusPlus::Pcap++)
+        message("Compiling Tutorial out of project")
+        find_package(PcapPlusPlus REQUIRED)
+        # We want to have the binary compiled in the same folder as the .cpp to be near the pcap file
+        set_target_properties("${PROJECT_NAME}" PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}")
+else()
+        set_target_properties("${PROJECT_NAME}" PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${PCAPPP_BINARY_TUTORIAL_DIR}")
```
Comment (seladb) [inline Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:12]: Building the DPDK tutorial as part of the project doesn't work, here the error I got:
```shell
/usr/bin/ld: CMakeFiles/Tutorial-DpdkL2Fwd.dir/main.cpp.o: in function `main':
main.cpp:(.text.startup+0x1eb): undefined reference to `L2FwdWorkerThread::L2FwdWorkerThread(pcpp::DpdkDevice*, pcpp::DpdkDevice*)'
/usr/bin/ld: main.cpp:(.text.startup+0x21b): undefined reference to `L2FwdWorkerThread::L2FwdWorkerThread(pcpp::DpdkDevice*, pcpp::DpdkDevice*)'
collect2: error: ld returned 1 exit status
make[2]: *** [Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeFiles/Tutorial-DpdkL2Fwd.dir/build.make:113: tutorials_bin/Tutorial-DpdkL2Fwd] Error 1
make[1]: *** [CMakeFiles/Makefile2:2339: Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeFiles/Tutorial-DpdkL2Fwd.dir/all] Error 2
```

File: Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt [hunk 4]
```diff
@@ -1,5 +1,15 @@
...
+project(Tutorial-DpdkL2Fwd)
 
-target_link_libraries(Tutorial-DpdkL2Fwd PUBLIC PcapPlusPlus::Pcap++)
+add_executable("${PROJECT_NAME}" main.cpp)
 
-set_target_properties(Tutorial-DpdkL2Fwd PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${PCAPPP_BINARY_TUTORIAL_DIR}")
+if(NOT TARGET PcapPlusPlus::Pcap++)
+        message("Compiling Tutorial out of project")
+        find_package(PcapPlusPlus REQUIRED)
+        # We want to have the binary compiled in the same folder as the .cpp to be near the pcap file
```
Comment (seladb) [inline Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:9]: Out of project also doesn't work:

```shell
usr/bin/ld: CMakeFiles/Tutorial-DpdkL2Fwd.dir/main.cpp.o: in function `printStats(pcpp::DpdkDevice*, pcpp::DpdkDevice*)':
main.cpp:(.text+0x96): undefined reference to `pcpp::DpdkDevice::getStatistics(pcpp::DpdkDevice::DpdkDeviceStats&) const'
/usr/bin/ld: main.cpp:(.text+0xaf): undefined reference to `pcpp::DpdkDevice::getStatistics(pcpp::DpdkDevice::DpdkDeviceStats&) const'
/usr/bin/ld: CMakeFiles/Tutorial-DpdkL2Fwd.dir/main.cpp.o: in function `main':
main.cpp:(.text+0x893): undefined reference to `pcpp::DpdkDeviceList::initDpdk(unsigned int, unsigned int, unsigned char, unsigned int, char**)'
/usr/bin/ld: main.cpp:(.text+0x8a5): undefined reference to `pcpp::DpdkDeviceList::getDeviceByPort(int) const'
/usr/bin/ld: main.cpp:(.text+0x910): undefined reference to `pcpp::DpdkDeviceList::getDeviceByPort(int) const'
/usr/bin/ld: main.cpp:(.text+0x97a): undefined reference to `pcpp::DpdkDevice::m_RSSKey'
/usr/bin/ld: main.cpp:(.text+0x9b8): undefined reference to `pcpp::DpdkDevice::openMultiQueues(unsigned short, unsigned short, pcpp::DpdkDevice::DpdkDeviceConfiguration const&)'
/usr/bin/ld: main.cpp:(.text+0xa6e): undefined reference to `pcpp::DpdkDevice::m_RSSKey'
/usr/bin/ld: main.cpp:(.text+0xaac): undefined reference to `pcpp::DpdkDevice::openMultiQueues(unsigned short, unsigned short, pcpp::DpdkDevice::DpdkDeviceConfiguration const&)'
/usr/bin/ld: main.cpp:(.text+0xb7d): undefined reference to `L2FwdWorkerThread::L2FwdWorkerThread(pcpp::DpdkDevice*, pcpp::DpdkDevice*)'
/usr/bin/ld: main.cpp:(.text+0xbb4): undefined reference to `L2FwdWorkerThread::L2FwdWorkerThread(pcpp::DpdkDevice*, pcpp::DpdkDevice*)'
/usr/bin/ld: main.cpp:(.text+0xc24): undefined reference to `pcpp::DpdkDeviceList::startDpdkWorkerThreads(unsigned int, std::vector<pcpp::DpdkWorkerThread*, std::allocator<pcpp::DpdkWorkerThread*> >&)'
/usr/bin/ld: main.cpp:(.text+0xe03): undefined reference to `pcpp::DpdkDeviceList::stopDpdkWorkerThreads()'
/usr/bin/ld: CMakeFiles/Tutorial-DpdkL2Fwd.dir/main.cpp.o: in function `pcpp::DpdkDeviceList::isInitialized() const':
main.cpp:(.text._ZNK4pcpp14DpdkDeviceList13isInitializedEv[_ZNK4pcpp14DpdkDeviceList13isInitializedEv]+0x1a): undefined reference to `pcpp::DpdkDeviceList::m_IsDpdkInitialized'
/usr/bin/ld: CMakeFiles/Tutorial-DpdkL2Fwd.dir/main.cpp.o: in function `pcpp::DpdkDeviceList::getInstance()':
main.cpp:(.text._ZN4pcpp14DpdkDeviceList11getInstanceEv[_ZN4pcpp14DpdkDeviceList11getInstanceEv]+0x3e): undefined reference to `pcpp::DpdkDeviceList::DpdkDeviceList()'
/usr/bin/ld: main.cpp:(.text._ZN4pcpp14DpdkDeviceList11getInstanceEv[_ZN4pcpp14DpdkDeviceList11getInstanceEv]+0x53): undefined reference to `pcpp::DpdkDeviceList::~DpdkDeviceList()'
/usr/bin/ld: main.cpp:(.text._ZN4pcpp14DpdkDeviceList11getInstanceEv[_ZN4pcpp14DpdkDeviceList11getInstanceEv]+0x80): undefined reference to `pcpp::DpdkDeviceList::m_MBufPoolSizePerDevice'
/usr/bin/ld: main.cpp:(.text._ZN4pcpp14DpdkDeviceList11getInstanceEv[_ZN4pcpp14DpdkDeviceList11getInstanceEv]+0x8e): undefined reference to `pcpp::DpdkDeviceList::initDpdkDevices(unsigned int)'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/Tutorial-DpdkL2Fwd.dir/build.make:88: ../Tutorial-DpdkL2Fwd] Error 1
make[1]: *** [CMakeFiles/Makefile2:76: CMakeFiles/Tutorial-DpdkL2Fwd.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

File: Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt (hunk 3 above)
Comment (clementperon) [reply Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:12]: should be fixed

Comment (seladb) [reply Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:12]: I can confirm it's fixed, thanks!

File: Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt (hunk 4 above)
Comment (clementperon) [reply Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:9]: Should be fixed now ?

Comment (seladb) [reply Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:9]: It still doesn't work for me and gives the same error 😕

Comment (clementperon) [reply Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:9]: Did you regenerate cmake ? And redo a cmake install ?

Comment (seladb) [reply Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:9]: yes...

Comment (clementperon) [reply Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:9]: Which dpdk/docker version ? I tried with docker 2111 and it worked fine

Comment (seladb) [reply Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:9]: Hmm maybe that's the issue... here is my setup:
- VM and not docker
- DPDK 18.11 installed in a non-default directory: `/opt/seladb/dpdk-stable-18.11.11/install`
- PcapPlusPlus installed in a non-default directory: `/opt/seladb/PcapPlusPlus/install`
- Built the `Tutorial-DpdkL2Fwd` tutorial with the following command: 
  - `CMAKE_PREFIX_PATH="/opt/seladb/PcapPlusPlus/install;/opt/seladb/dpdk-stable-18.11.11/install" cmake -S . -B build` (succeeds)
  - `cmake --build build` (fails with the error above)

Comment (clementperon) [reply Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:9]: what the output of cmake at the configure time ?

Comment (seladb) [reply Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:9]: I can confirm it works with DPDK 21.11 if I installed it in the default directories.
But it doesn't work with DPDK 18.11 installed in a non-default directory.

Here is the full output of cmake at configure and build time:
```shell
$ CMAKE_PREFIX_PATH="/opt/seladb/PcapPlusPlus/install/;/opt/seladb/dpdk-stable-18.11.11/install/" cmake -S . -B build
-- The C compiler identification is GNU 9.4.0
-- The CXX compiler identification is GNU 9.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
Compiling Tutorial out of project
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Found PCAP: /usr/local/lib/libpcap.so  
-- Performing Test PCAP_LINKS_SOLO
-- Performing Test PCAP_LINKS_SOLO - Success
-- Looking for pcap_set_immediate_mode
-- Looking for pcap_set_immediate_mode - found
-- Looking for pcap_setdirection
-- Looking for pcap_setdirection - found
-- Configuring done
-- Generating done
-- Build files have been written to: /opt/seladb/PcapPlusPlus/Examples/Tutorials/Tutorial-DpdkL2Fwd/build

seladb@ubunu2004 /opt/seladb/PcapPlusPlus/Examples/Tutorials/Tutorial-DpdkL2Fwd build_tutorials_out_of_project *1 ?15
$ cmake --build build
Scanning dependencies of target Tutorial-DpdkL2Fwd
[ 33%] Building CXX object CMakeFiles/Tutorial-DpdkL2Fwd.dir/main.cpp.o
[ 66%] Building CXX object CMakeFiles/Tutorial-DpdkL2Fwd.dir/WorkerThread.cpp.o
[100%] Linking CXX executable ../Tutorial-DpdkL2Fwd
/usr/bin/ld: CMakeFiles/Tutorial-DpdkL2Fwd.dir/main.cpp.o: in function `printStats(pcpp::DpdkDevice*, pcpp::DpdkDevice*)':
main.cpp:(.text+0x96): undefined reference to `pcpp::DpdkDevice::getStatistics(pcpp::DpdkDevice::DpdkDeviceStats&) const'
/usr/bin/ld: main.cpp:(.text+0xaf): undefined reference to `pcpp::DpdkDevice::getStatistics(pcpp::DpdkDevice::DpdkDeviceStats&) const'
/usr/bin/ld: CMakeFiles/Tutorial-DpdkL2Fwd.dir/main.cpp.o: in function `main':
main.cpp:(.text+0x893): undefined reference to `pcpp::DpdkDeviceList::initDpdk(unsigned int, unsigned int, unsigned char, unsigned int, char**)'
/usr/bin/ld: main.cpp:(.text+0x8a5): undefined reference to `pcpp::DpdkDeviceList::getDeviceByPort(int) const'
/usr/bin/ld: main.cpp:(.text+0x910): undefined reference to `pcpp::DpdkDeviceList::getDeviceByPort(int) const'
/usr/bin/ld: main.cpp:(.text+0x97a): undefined reference to `pcpp::DpdkDevice::m_RSSKey'
/usr/bin/ld: main.cpp:(.text+0x9b8): undefined reference to `pcpp::DpdkDevice::openMultiQueues(unsigned short, unsigned short, pcpp::DpdkDevice::DpdkDeviceConfiguration const&)'
/usr/bin/ld: main.cpp:(.text+0xa6e): undefined reference to `pcpp::DpdkDevice::m_RSSKey'
/usr/bin/ld: main.cpp:(.text+0xaac): undefined reference to `pcpp::DpdkDevice::openMultiQueues(unsigned short, unsigned short, pcpp::DpdkDevice::DpdkDeviceConfiguration const&)'
/usr/bin/ld: main.cpp:(.text+0xc24): undefined reference to `pcpp::DpdkDeviceList::startDpdkWorkerThreads(unsigned int, std::vector<pcpp::DpdkWorkerThread*, std::allocator<pcpp::DpdkWorkerThread*> >&)'
/usr/bin/ld: main.cpp:(.text+0xe03): undefined reference to `pcpp::DpdkDeviceList::stopDpdkWorkerThreads()'
/usr/bin/ld: CMakeFiles/Tutorial-DpdkL2Fwd.dir/main.cpp.o: in function `pcpp::DpdkDeviceList::isInitialized() const':
main.cpp:(.text._ZNK4pcpp14DpdkDeviceList13isInitializedEv[_ZNK4pcpp14DpdkDeviceList13isInitializedEv]+0x1a): undefined reference to `pcpp::DpdkDeviceList::m_IsDpdkInitialized'
/usr/bin/ld: CMakeFiles/Tutorial-DpdkL2Fwd.dir/main.cpp.o: in function `pcpp::DpdkDeviceList::getInstance()':
main.cpp:(.text._ZN4pcpp14DpdkDeviceList11getInstanceEv[_ZN4pcpp14DpdkDeviceList11getInstanceEv]+0x3e): undefined reference to `pcpp::DpdkDeviceList::DpdkDeviceList()'
/usr/bin/ld: main.cpp:(.text._ZN4pcpp14DpdkDeviceList11getInstanceEv[_ZN4pcpp14DpdkDeviceList11getInstanceEv]+0x53): undefined reference to `pcpp::DpdkDeviceList::~DpdkDeviceList()'
/usr/bin/ld: main.cpp:(.text._ZN4pcpp14DpdkDeviceList11getInstanceEv[_ZN4pcpp14DpdkDeviceList11getInstanceEv]+0x80): undefined reference to `pcpp::DpdkDeviceList::m_MBufPoolSizePerDevice'
/usr/bin/ld: main.cpp:(.text._ZN4pcpp14DpdkDeviceList11getInstanceEv[_ZN4pcpp14DpdkDeviceList11getInstanceEv]+0x8e): undefined reference to `pcpp::DpdkDeviceList::initDpdkDevices(unsigned int)'
/usr/bin/ld: CMakeFiles/Tutorial-DpdkL2Fwd.dir/WorkerThread.cpp.o: in function `L2FwdWorkerThread::run(unsigned int)':
WorkerThread.cpp:(.text+0xec): undefined reference to `pcpp::DpdkDevice::receivePackets(pcpp::MBufRawPacket**, unsigned short, unsigned short) const'
/usr/bin/ld: WorkerThread.cpp:(.text+0x129): undefined reference to `pcpp::DpdkDevice::sendPackets(pcpp::MBufRawPacket**, unsigned short, unsigned short, bool)'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/Tutorial-DpdkL2Fwd.dir/build.make:103: ../Tutorial-DpdkL2Fwd] Error 1
make[1]: *** [CMakeFiles/Makefile2:76: CMakeFiles/Tutorial-DpdkL2Fwd.dir/all] Error 2
make: *** [Makefile:84: all] Error 2

```

Comment (clementperon) [reply Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:9]: strange cmake doesn't even look for DPDK... 
can you add
```
set(CMAKE_FIND_DEBUG_MODE TRUE)
find_package(PcapPlusPlus REQUIRED)
set(CMAKE_FIND_DEBUG_MODE FALSE)
```
and redo a configuration please ?

Comment (clementperon) [reply Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:9]: ha or maybe the find_dependency(DPDK) is silent and you need to set:
-DDPDK_ROOT=<DPDK_PATH> when doing the CMake configuration of the DPDKL2FWD

Comment (seladb) [reply Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:9]: I added the debug lines, here is the output:

```shell
$ CMAKE_PREFIX_PATH="/opt/seladb/PcapPlusPlus/install/;/opt/seladb/dpdk-stable-18.11.11/install/" cmake -S . -B build
-- The C compiler identification is GNU 9.4.0
-- The CXX compiler identification is GNU 9.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
Compiling Tutorial out of project
Checking prefix [/opt/seladb/PcapPlusPlus/install/;/opt/seladb/dpdk-stable-18.11.11/install/]
Checking prefix [/home/linuxbrew/.linuxbrew/]
Checking file [/home/linuxbrew/.linuxbrew/PcapPlusPlusConfig.cmake]
Checking file [/home/linuxbrew/.linuxbrew/pcapplusplus-config.cmake]
Checking prefix [/home/seladb/.cargo/]
Checking file [/home/seladb/.cargo/PcapPlusPlusConfig.cmake]
Checking file [/home/seladb/.cargo/pcapplusplus-config.cmake]
Checking prefix [/home/seladb/.local/]
Checking file [/home/seladb/.local/PcapPlusPlusConfig.cmake]
Checking file [/home/seladb/.local/pcapplusplus-config.cmake]
Checking prefix [/usr/local/]
Checking file [/usr/local/PcapPlusPlusConfig.cmake]
Checking file [/usr/local/pcapplusplus-config.cmake]
Checking file [/usr/local/lib/cmake/pcapplusplus/PcapPlusPlusConfig.cmake]
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Found PCAP: /usr/local/lib/libpcap.so  
-- Performing Test PCAP_LINKS_SOLO
-- Performing Test PCAP_LINKS_SOLO - Success
-- Looking for pcap_set_immediate_mode
-- Looking for pcap_set_immediate_mode - found
-- Looking for pcap_setdirection
-- Looking for pcap_setdirection - found
-- Configuring done
-- Generating done
-- Build files have been written to: /opt/seladb/PcapPlusPlus/Examples/Tutorials/Tutorial-DpdkL2Fwd/build
```

Comment (seladb) [reply Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:9]: > ha or maybe the find_dependency(DPDK) is silent and you need to set: -DDPDK_ROOT=<DPDK_PATH> when doing the CMake configuration of the DPDKL2FWD

This doesn't work and provides the following warning:
```shell
CMake Warning:
  Manually-specified variables were not used by the project:

    DPDK_ROOT
```

Comment (clementperon) [reply Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:9]: Strange it seems that you didn't compile PcapPP with DPDK support or that you have another PcapPlusPlusConfig.cmake on your system.

Coud you check a find -name "PcapPlusPlusConfig.cmake" / ?
and display the content of the files

Comment (seladb) [reply Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:9]: I'm almost sure I compiled PcapPlusPlus with DPDK, here is the output of the file:

```shell
seladb@ubunu2004 /opt/seladb/PcapPlusPlus build_tutorials_out_of_project *1 !1 ?14
$ find . -name "PcapPlusPlusConfig.cmake"
./build/PcapPlusPlusConfig.cmake
./install/lib/cmake/pcapplusplus/PcapPlusPlusConfig.cmake
seladb@ubunu2004 /opt/seladb/PcapPlusPlus build_tutorials_out_of_project *1 !1 ?14
$ cat ./build/PcapPlusPlusConfig.cmake
# ~~~
# - Config file for the PcapPlusPlus package
# It defines the following variables
#  PcapPlusPlus_INCLUDE_DIRS - include directories for PcapPlusPlus
#  PcapPlusPlus_LIBRARIES    - libraries to link against
# ~~~


####### Expanded from @PACKAGE_INIT@ by configure_package_config_file() #######
####### Any changes to this file will be overwritten by the next CMake run ####
####### The input file was PcapPlusPlusConfig.cmake.in                            ########

get_filename_component(PACKAGE_PREFIX_DIR "${CMAKE_CURRENT_LIST_DIR}/../../../" ABSOLUTE)

macro(set_and_check _var _file)
  set(${_var} "${_file}")
  if(NOT EXISTS "${_file}")
    message(FATAL_ERROR "File or directory ${_file} referenced by variable ${_var} does not exist !")
  endif()
endmacro()

macro(check_required_components _NAME)
  foreach(comp ${${_NAME}_FIND_COMPONENTS})
    if(NOT ${_NAME}_${comp}_FOUND)
      if(${_NAME}_FIND_REQUIRED_${comp})
        set(${_NAME}_FOUND FALSE)
      endif()
    endif()
  endforeach()
endmacro()

####################################################################################

list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/modules")

include(CMakeFindDependencyMacro)

set_and_check(PcapPlusPlus_INCLUDE_DIR
              ${PACKAGE_PREFIX_DIR}/include/pcapplusplus)

if(UNIX)
  set(THREADS_PREFER_PTHREAD_FLAG ON)
endif()
find_dependency(Threads)

list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}")

find_dependency(PCAP)

find_dependency(DPDK)


include("${CMAKE_CURRENT_LIST_DIR}/PcapPlusPlusTargets.cmake")

check_required_components(PcapPlusPlus)
```

Here is how I compiled PcapPlusPlus:
```shell
cmake -DPCAPPP_USE_DPDK=ON -DDPDK_ROOT=/opt/seladb/dpdk-stable-18.11.11/install -S . -B build
```

Comment (clementperon) [reply Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:9]: you should make a "find / -name "PcapPlusPlusConfig.cmake" the cmake should be in /usr/lib/cmake/xxx

Comment (clementperon) [reply Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:9]: but maybe you already installed without DPDK and didn't installed the new version

Comment (seladb) [reply Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:9]: I tried to build PcapPlusPlus with a DPDK version that is installed in `/opt/seladb/dpdk-stable-18.11.11/install`

Comment (seladb) [reply Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:9]: Maybe it's just an issue with my specific setup. It worked with DPDK 21.11 which is installed in the default directories (`/usr/local/xxx`)

Comment (seladb) [reply Examples/Tutorials/Tutorial-DpdkL2Fwd/CMakeLists.txt:9]: I think we can merge it as is, should be fine

**Review by seladb — APPROVED** [review]

**Pull request merged**